e2e/main.e2e.spec.ts

```diff
@@ -0,0 +1,84 @@
+import path from 'path';
+import os from 'os';
+import { Application, SpectronClient } from 'spectron';
+import { BrowserWindow } from 'electron';
+import NavBarPage from './pages/navbar';
+import GeneralPage from './pages/general';
+import KubernetesPage from './pages/kubernetes';
+import PortForwardingPage from './pages/portforwarding';
+import ImagesPage from './pages/images';
+import TroubleshootingPage from './pages/troubleshooting';
+const electronPath = require('electron');
+
+jest.setTimeout(60_000);
+
+describe('Rancher Desktop', () => {
+  let app: Application;
+  let client: SpectronClient;
+  let browserWindow: BrowserWindow;
+  let navBarPage: NavBarPage;
+
+  beforeAll(async() => {
+    app = new Application({
+      // 'any' typing is required for now as other alternate usage/import
+      //  cause issues running the tests. Without 'any' typescript
+      //  complains of type mismatch.
+      path: electronPath as any,
+      args: [path.dirname(__dirname)],
+      env:  { SPECTRON_RUN: 'yes' }
+    });
+
+    await app.start();
+    client = app.client;
+    browserWindow = app.browserWindow;
+    navBarPage = new NavBarPage(app);
+  });
+
+  afterAll(async() => {
+    if (app && app.isRunning()) {
+      await app.stop();
+    }
+  });
+
+  it('opens the window', async() => {
+    await client.waitUntilWindowLoaded();
+    // typescript doesn't see a value of await in below statement, but
+    // removing await makes the statement not wait till the app window loads
+    // Also, Alternate ways to get the app window title, for example using client
+    // didn't work. So, Leaving 'await' for now. We may need to review this and
+    // fix this in future.
+    const title = await browserWindow.getTitle();
+
+    expect(title).toBe('Rancher Desktop');
+  });
+
+  it('should display welcome message in general tab', async() => {
+    const generalPage: GeneralPage = await navBarPage.getGeneralPage();
+
+    expect(await generalPage.getTitle()).toBe('Welcome to Rancher Desktop');
+  });
+
+  it('should switch to Kubernetes Settings tab', async() => {
+    const kubernetesPage: KubernetesPage = await navBarPage.getKubernetesPage();
+
+    expect(await kubernetesPage.getResetKubernetesButtonText()).toBe('Reset Kubernetes');
+  });
+
+  it('should switch to Port Forwarding tab', async() => {
+    const portForwardingPage: PortForwardingPage = await navBarPage.getPortForwardingPage();
+
+    os.platform().startsWith('win') ? expect(portForwardingPage).not.toBeNull() : expect(portForwardingPage).toBeNull();
```

Comment:
This doesn't seem to work:
`getPortForwardingPage` always returns an object (`return new PortForwardingPage…`), regardless of whether the result was found.

Running the test, I get:
```
  ● Rancher Desktop › should switch to Port Forwarding tab

    expect(received).toBeNull()

    Received: {…}
      68 |     const portForwardingPage: PortForwardingPage = await navBarPage.getPortForwardingPage();
      69 |
    > 70 |     os.platform().startsWith('win') ? expect(portForwardingPage).not.toBeNull() : expect(portForwardingPage).toBeNull();
         |                                                                                                              ^
      71 |   });
      72 |
      73 |   it('should switch to Images tab', async() => {

      at Object.<anonymous> (e2e/main.e2e.spec.ts:70:110)

  ● Rancher Desktop › should switch to Images tab

    Can't call click on element with selector ".nav li[item="/PortForwarding"] a" because element wasn't found

      18 |
      19 |     async clickOnNavBarItem(item: string) {
    > 20 |       await (await this.client.$(`.nav li[item="/${ item }"] a`)).click();
         |       ^
      21 |       await this.client.waitUntilWindowLoaded(60_000);
      22 |     }
      23 |

      at Object.implicitWait [as default] (node_modules/webdriverio/build/utils/implicitWait.js:26:19)
      at Element.elementErrorHandlerCallbackFn (node_modules/webdriverio/build/middlewares.js:13:29)
      at Element.wrapCommandFn (node_modules/@wdio/utils/build/shim.js:63:29)
      at NavBarPage.clickOnNavBarItem (e2e/pages/navbar.ts:20:7)
      at NavBarPage.getImagesPage (e2e/pages/navbar.ts:37:7)
      at Object.<anonymous> (e2e/main.e2e.spec.ts:74:36)
```